web/src/main/java/com/epam/esm/web/controller/OrderController.java

```diff
@@ -30,23 +33,28 @@ public CollectionModel<OrderModel> getOrders(
     }
 
     @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ADMIN')")
     public OrderModel getOrder(@PathVariable("id") Long id) {
         return orderModelAssembler.toModel(orderService.findById(id));
     }
 
     @PostMapping
-    public OrderModel createOrder(@RequestBody @Valid OrderDTO dto) {
-        Order createdOrder = orderService.create(dto);
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
+    public OrderModel createOrder(@RequestBody @Valid OrderDTO dto, Authentication authentication) {
+
+        Order createdOrder = orderService.create(dto, authentication);
         return orderModelAssembler.toModel(createdOrder);
     }
 
     @GetMapping("/{userId}/user-orders")
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
     public CollectionModel<OrderModel> getOrdersByUserId(
```

Comment:
This endpoint is a bit unclear for me. In method arguments, we can see user id and authentication also. Which one is being used to retrieve data? I would advise creating separate endpoints for admin and users also